src/edu/csus/ecs/pc2/core/ParseArguments.java

```diff
@@ -301,6 +305,24 @@ public boolean isOptPresent(String argKey) {
         return obj != null; 
     }
 
+    /**
+     * Does the value match any of the allowed options?
+     * 
+     * @param value
+     * @return boolean
+     */
+    public boolean isAllowedOption(String value) {
+        if(allowedOptions != null) {
+            for(String sOpt: allowedOptions) {
+                if(sOpt.equals(value)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+        return true;
```

Comment:
Is it really correct to say that if `allowedOptions` is null, the input `value` is an "allowed option"?

Reply:
Yes.  If the allowedOptions array is not set, then all options are allowed and no checking is done.  This is the default behavior to maintain compatibility with other classes that use this class.  I clarified the comment for the isAllowedOption method.